javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentClassLoader.java

```diff
@@ -92,6 +140,89 @@ public AgentClassLoader(
     return super.loadClass(name, resolve);
   }
 
+  @Override
+  protected Class<?> findClass(String name) throws ClassNotFoundException {
+    JarEntry jarEntry = findJarEntry(name.replace('.', '/') + ".class");
```

Comment:
Do you change the usual "parent-first" load order on purpose? Why?

Reply:
Parent delegation is handled in `loadClass` and we use the default parent first strategy. Here I first search for the class in `inst/` and after that call `super.findClass` to find the class from agent initializer jar.
As a side note changing delegation to child first might be a good idea as currently when somebody adds something that we use in agent to bootclasspath we'll use it instead of what is in the agent. I guess this could be a feature or a bug depending on how you look it.